siege_utilities/geo/grids.py

```
@@ -0,0 +1,187 @@
+        from .h3_utils import h3_index_points
+        if level is not None and resolution is None:
+            # Caller asked for grid='h3' but used 'level'; honour it.
+            resolution = level
+        if kwargs:
+            # Reject silent kwarg dropping — symmetric with the S2 path,
+            # which forwards them. h3_index_points has no extra kwargs to
+            # forward, so anything here is caller error.
+            raise TypeError(
+                f"index_points(grid='h3') got unexpected keyword argument(s): "
+                f"{sorted(kwargs)}"
+            )
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)